httplib.h

```diff
@@ -7584,6 +7587,39 @@ inline void ClientImpl::set_ca_cert_store(X509_STORE *ca_cert_store) {
     ca_cert_store_ = ca_cert_store;
   }
 }
+
+inline X509_STORE *ClientImpl::create_ca_cert_store(const char *ca_cert,
+                                                    std::size_t size) {
+  BIO *mem = nullptr;
+  STACK_OF(X509_INFO) *inf = nullptr;
+  X509_STORE *cts = nullptr;
+
+  [&] {
+    mem = BIO_new_mem_buf(ca_cert, size);
+    if (!mem) return;
+
+    inf = PEM_X509_INFO_read_bio(mem, nullptr, nullptr, nullptr);
+    if (!inf) return;
+
+    cts = X509_STORE_new();
+    if (!cts) return;
+
+    for (int first = 0, last = sk_X509_INFO_num(inf); first < last; ++first) {
+      auto *itmp = sk_X509_INFO_value(inf, first);
```

Comment:
@yhirose Yes, it could be because both deduce the same type. I used `auto *` to explicitly express the return type is pointer. And it also throws an error when the return type is changed from a pointer type.
Actually I don't have a strong opinion on whether it is `auto` or `auto *`. This is just a personal style or preference.

Reply:
Thanks for the replay. Could you change it to just `auto`?

Reply:
Sure